### PR TITLE
Update C++ implementations for Verlet algorithms

### PR DESCRIPTION
**Pull request by berquist**
Done according to https://github.com/algorithm-archivists/algorithm-archive/issues/257.

### CODE REVIEW

File: contents/verlet_integration/code/c++/verlet.cpp [hunk 1]
```diff
@@ -45,18 +48,41 @@ double velocity_verlet(double pos, double acc, double dt) {
     vel += acc * dt;
   }
 
-  return time;
+  return std::make_pair(time, vel);
 }
 
 int main() {
+  double time, vel;
```
Comment (Gathros) [inline contents/verlet_integration/code/c++/verlet.cpp]: Why not just print straight to console.

File: contents/verlet_integration/code/c++/verlet.cpp [hunk 2]
```diff
@@ -1,6 +1,9 @@
+#include <iomanip>
 #include <iostream>
+#include <utility>
+
+typedef std::pair<double, double> vpair;
```
Comment (antonte) [inline contents/verlet_integration/code/c++/verlet.cpp]:
``` C++
using vpair = std::pair<double, double>;
```
is preferable way.

[T.43: Prefer using over typedef for defining](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#t43-prefer-using-over-typedef-for-defining-aliases)

File: contents/verlet_integration/code/c++/verlet.cpp (hunk 1 above)
Comment (antonte) [inline contents/verlet_integration/code/c++/verlet.cpp]: [ES.20: Always initialize an object](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es20-always-initialize-an-object)

File: contents/verlet_integration/code/c++/verlet.cpp [hunk 3]
```diff
@@ -45,18 +48,41 @@ double velocity_verlet(double pos, double acc, double dt) {
...
+  time = time_vel_pair.first;
+  vel = time_vel_pair.second;
+  std::cout << "Time for Stormer Verlet integration is: " \
+            << time << std::endl;
+  std::cout << "Velocity for Stormer Verlet integration is: " \
+            << vel << std::endl;
 
-  // Note that depending on the simulation, you might want to have the verlet
-  // loop outside.
+  time_vel_pair = velocity_verlet(5.0, -10, 0.01);
```
Comment (antonte) [inline contents/verlet_integration/code/c++/verlet.cpp]: you can use [```std::tie```](https://en.cppreference.com/w/cpp/utility/tuple/tie)

File: contents/verlet_integration/code/c++/verlet.cpp [hunk 4]
```diff
@@ -16,8 +19,7 @@ double verlet(double pos, double acc, double dt) {
   return time;
 }
 
-// Simple function for stormer-verlet
-double stormer_verlet(double pos, double acc, double dt) {
+std::pair<double, double> stormer_verlet(double pos, double acc, double dt) {
```
Comment (antonte) [inline contents/verlet_integration/code/c++/verlet.cpp]: use ```vpair```

File: contents/verlet_integration/code/c++/verlet.cpp [hunk 5]
```diff
@@ -28,14 +30,15 @@ double stormer_verlet(double pos, double acc, double dt) {
...
+    // straightforward
     vel += acc * dt;
   }
 
-  return time;
+  return std::make_pair(time, vel);
 }
 
-double velocity_verlet(double pos, double acc, double dt) {
+std::pair<double, double> velocity_verlet(double pos, double acc, double dt) {
```
Comment (antonte) [inline contents/verlet_integration/code/c++/verlet.cpp]: use ```vpair```

File: contents/verlet_integration/code/c++/verlet.cpp (hunk 2 above)
Comment (Gustorn) [reply contents/verlet_integration/code/c++/verlet.cpp]: I think I would prefer a simple struct with descriptive fields over the pair anyway. `std::pair` is alright for utility types but I dislike it for types that should have some semantic meaning.

File: contents/verlet_integration/code/c++/verlet.cpp [hunk 6]
```diff
@@ -45,18 +50,35 @@ double velocity_verlet(double pos, double acc, double dt) {
...
+  // For example, if your acceleration chages as a function of time,
+  // you might need to also change the acceleration to be read into
+  // each of these functions.
+
+  double time = verlet(5.0, -10, 0.01);
+  std::cout << "Time for Verlet integration is: " \
+            << time << std::endl;
+
+  timestep timestep_sv = stormer_verlet(5.0, -10, 0.01);
+  std::cout << "Time for Stormer Verlet integration is: " \
```
Comment (Gathros) [inline contents/verlet_integration/code/c++/verlet.cpp]: Why is there a backslash?

Comment (berquist) [reply contents/verlet_integration/code/c++/verlet.cpp]: I prefer to use an explicit line continuation indicator even when not required.

**Pull request merged**